akkeeper/src/main/scala/akkeeper/master/route/ControllerComposite.scala

```diff
@@ -15,16 +15,11 @@
  */
 package akkeeper.master.route
 
-import akka.http.scaladsl.server.{PathMatchers, Route}
-import ControllerComposite._
+import akka.http.scaladsl.server.{Directives, PathMatchers, Route}
 
 case class ControllerComposite(basePath: String,
-                               controllers: Seq[BaseController]) extends BaseController {
-  override val route: Route = pathPrefix(PathMatchers.separateOnSlashes(basePath)) {
-    controllers.reduceLeft((a, b) => UnitController(a.route ~ b.route)).route
+                               controllers: Seq[BaseController]) extends Directives {
```

Comment:
I'd still prefer the `ControllerComposite` to share the same trait as other controllers because of the `def route(): Route` interface (and it's still a controller, isn't it?). Can we adopt the following hierarchy to make all controllers happy and preserve the shared interface at the same time?:
```
trait BaseController extends Directives with SprayJsonSupport {
  def route: Route
}

trait RequestHandlingController extends BaseController with CommonApiJsonProtocol {
  protected def service: ActorRef 

  // all the handler related logic
}
```
This way `ControllerComposite` can still inherit `BaseController`. It'll be great if you have a better name for `RequestHandlingController` :)